accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,191 @@
+
+### Inplace servicing updates are good enough for the compiler, why not source generators?
+
+The compiler has a very stable 1:1 relationship between a user's source and the code it generates. As such the user can control the output of the compiler through their own usage.  The compiler's contribution of complexity to the user's assembly is very small and highly specified.
+ 
+Contrast this to source generators which can take a very small amount of user source to produce a large amount of generated code.  This large amount of generated code contains significant complexity which is loosely specified.  Its behavior is specific to the component it extends.  It's likely that we'd need to bugfix this code over time and change it as we develop features in the components it extends.  It is also likely that we will learn new techniques and improvments to source generators over time, increasing the changes for bugfixes and features.
+
+Users expect stability for projects which do not change, for example a project targeting an LTS release. It is much more difficult to ensure this stability when we need to significantly components which contribute complexity to that project.
+
+### Isn't this a lot of servicing if we find a bug?
+
+Although this seems like more changes than if behavior was in the compiler-distributed analyzers it is not.  If we cared enough about a bug to service an analyzer that runs on a stable .NET version we would also want to service the SDK for that stable .NET Version: including the same number of changes and branches.  In fact, this strategy *reduces* servicing in the case where a bug only exists in a single version of source generator.
+
+What this strategy does do is reduce risk.  We eliminate the channel where the latest SDK would contain changes that impacted users' projects targeting older .NET versions.  That is a good thing as it reduces the need to test those configurations and ensures we can garuntee the quality and stability of our stable .NET releases.
```

Comment:
~~garuntee~~ -> guarantee

Reply:
Thanks.  I will never learn the correct spelling of this word.  🤦